test/capture_test.exs

```diff
@@ -264,6 +278,29 @@ defmodule CaptureTest do
                      end
       end)
     end
+
+    test "initially deferred trigger allows late transaction insertion" do
+      transaction_with_simulated_commit(fn ->
+        # deferred_rabbits have a trigger with INITIALLY DEFERRED constraint, so we can insert
+        # a record before inserting the transaction.
+        insert_deferred_jack()
+        insert_transaction()
+      end)
+
+      assert [%{"table_name" => "deferred_rabbits"}] = select_changes()
+    end
+
+    test "initially deferred trigger still requires a transaction to be inserted" do
+      msg =
+        "ERROR 23503 (foreign_key_violation) INSERT on table public.deferred_rabbits " <>
+          "without prior INSERT into carbonite_default.transactions"
+
+      assert_raise Postgrex.Error, msg, fn ->
+        transaction_with_simulated_commit(fn ->
+          insert_deferred_jack()
+        end)
+      end
+    end
```

Comment:
It is probably clear, but maybe refuting no rabbits were created would be good here, right? Since the transaction is rolled back.

Reply:
:shrug: can do